Update guidance for Portable PDB in 4.7.1

noahfalk opened the pull request.

noahfalk commented:
@rpetrusha @vivmishra @tmat - PTAL

rpetrusha commented on releases/net471/KnownIssues/517815-BCL Stack traces are missing source information for frames with debug information in the Portable PDB format.md:
```diff
@@ -0,0 +1,19 @@
+## Symptoms
+
+An application that formats stack traces is missing source information for some or all frames. This includes stack traces formatted via [System.Exception.ToString()](https://docs.microsoft.com/en-us/dotnet/api/system.exception.tostring?view=netframework-4.7.1), [System.Exception.StackTrace](https://docs.microsoft.com/en-us/dotnet/api/system.exception.stacktrace?view=netframework-4.7.1) and [Syste ...
+The frames missing source information must reside in assemblies that have pdbs in the [Portable PDB format](https://github.com/dotnet/core/blob/master/Documentation/diagnostics/portable_pdb.md) present on disk.
```
Since you're describing symptoms here, "must" is not necessary.

noahfalk replied:
My goal here was to qualify when you might be observing this issue vs. some other issue. For example you could build assemblies with any format pdb and then not put pdb on disk and you would observe the same missing source lines. However in that case you aren't hitting this issue, you are just failing to deploy your PDBs. I wanted to help people avoid some likely misdiagnosis.

Do you still I think I should remove 'must' or maybe phrase this a different way? I considered phrasing it as 'you observe _symptom_ when _condition_ is true' but it made a very long sentence whereas "must" helped me break the sentence apart. "You observe _symptom_. _condition_ must be met."

rpetrusha replied:
Basically, symptoms defines the two conditions that must be present for this issue to apply. So in that sense, the "must" is not necessary. So the intent is already clear. I think the sentence reads much better with it removed. But I'll leave it up to you to decide, @noahfalk.

rpetrusha commented on releases/net471/KnownIssues/517815-BCL Stack traces are missing source information for frames with debug information in the Portable PDB format.md:
```diff
@@ -0,0 +1,19 @@
+
+## Cause
+
+The .NET Framework 4.7.1 added support for detecting and parsing the Portable PDB file format to show file and line number information in stack traces. However due to an unfortunate [implementation issue](https://github.com/Microsoft/dotnet/blob/master/releases/net471/KnownIssues/517815-BCL%20Applications%20making%20heavy%20use%20of%20System.Diagnostics.StackTrace%20might%20run%20more%20slowly%20 ...
```
commas after:
- However 
- implementation issue
- customers
impacts can't be introduced; better: "this feature had an unacceptable performance impact"

vivmishra replied:
Remove "unfortunate"

rpetrusha commented on releases/net471/KnownIssues/517815-BCL Stack traces are missing source information for frames with debug information in the Portable PDB format.md:
```diff
@@ -0,0 +1,19 @@
+
+## Workarounds
+
+- If you are in control of the build process for the problematic assemblies you may be able to configure it to generate the classic Windows PDB format instead.
```
If you are in control --> If you control
comma after assemblies

vivmishra commented on releases/net471/KnownIssues/517815-BCL Stack traces are missing source information for frames with debug information in the Portable PDB format.md:
```diff
@@ -0,0 +1,19 @@
+# Stack traces are missing source information for frames with debug information in the Portable PDB format when running on .Net Framework 4.7.1
```
Make all instances of ".Net" --> ".NET"

vivmishra commented on releases/net471/KnownIssues/517815-BCL Stack traces are missing source information for frames with debug information in the Portable PDB format.md:
```diff
@@ -0,0 +1,19 @@
+
+## Resolution
+
+A fix is anticipated for .Net Framework 4.7.2 that restores Portable PDB functionality with greatly improved performance.
```
Suggestion: Replace 'for' --> 'in': A fix is anticipated in .NET Framework 4.7.2, in near future, that restores ... ...

Early access for 4.7.2 goes live on 2/5 -- we can then append the [reference](https://github.com/Microsoft/dotnet-framework-early-access/blob/master/README.md) to it and suggest folks to go try the product for feature again. 

The PR is https://github.com/Microsoft/dotnet-framework-early-access/pull/10

noahfalk commented:
Thanks to you both!

I've updated the PR with most of the feedback but I did have an outstanding question for Ron on "must". Ron if my explanation above makes sense I'm happy to have you apply whatever adjustment seems appropriate.

On consultation with Tomas I removed the workaround for using the converter and we'll put it back once he has documentation in better shape.

noahfalk commented:
"Must" is out of there : )

noahfalk commented:
I don't have permission to merge though.
